forge/src/runner.rs

```diff
@@ -210,251 +186,234 @@ impl<'a, DB: DatabaseRef + Send + Sync> ContractRunner<'a, DB> {
         self.executor.set_nonce(self.sender, 1);
 
         // Deploy libraries
-        self.predeploy_libs.iter().for_each(|code| {
-            self.executor
-                .deploy(*CALLER, code.0.clone(), 0u32.into())
-                .expect("couldn't deploy library");
-        });
+        let mut traces: Vec<(TraceKind, CallTraceArena)> = self
+            .predeploy_libs
+            .iter()
+            .filter_map(|code| {
+                let DeployResult { traces, .. } = self
```

Comment:
We need the traces for the deployment of the libraries and the test contract itself to be able to identify them later. The way we did this before was by passing around the executor in the identify step in case we needed to fetch contract code on the fly

Reply:
wait if we dont pass the evm into the `update_identified` any contracts deployed in setup wont be identified (or any contract deployed *inside* the call for that matter). I am fine with this, but just FYI that is very much needed to on the fly identify contracts by runtime bytecode.

Reply:
ooooh nice - we use the output field of the create traces to identify - nice